### PR TITLE
2TAP Feedback flow

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Fix https://github.com/mapbox/navigation-sdks/issues/333
Adding support for 2TAP Feedback flow

TODO:
* There is still one thing left that need to update the Telemetry related API to enable submitting Feedback Descriptions. Per @abhishek1508 , the telemetry SDK is not ready yet.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

Allow the user to use 2TAP Feedback flow to give more detail about the Navigation feedback.

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java [hunk 1]
```diff
@@ -42,26 +51,43 @@
 public class FeedbackBottomSheet extends BottomSheetDialogFragment implements Animator.AnimatorListener {
 
   public static final String TAG = FeedbackBottomSheet.class.getSimpleName();
+  public static final int FEEDBACK_MAIN_FLOW = 0;
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java:54]: may be make `@IntDef` for `FEEDBACK_MAIN_FLOW` and `FEEDBACK_DETAIL_FLOW`

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDescriptionItem.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package com.mapbox.navigation.ui.feedback;
+
+import androidx.annotation.StringRes;
+
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent;
+
+class FeedbackDescriptionItem {
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDescriptionItem.java:7]: should we write new code with Kotlint? I think it reasonable

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDescriptionViewHolder.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.mapbox.libnavigation.ui.R;
+
+class FeedbackDescriptionViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDescriptionViewHolder.java:11]: https://github.com/mapbox/mapbox-navigation-android/pull/2873/files#r419354567

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java (hunk 1 above)
Comment (JunDai) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java]: Fixed

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java [hunk 4]
```diff
@@ -287,8 +331,145 @@ public void onFeedbackItemClick(ImageView view, int feedbackPosition) {
...
+          FeedbackEvent.ROAD_IS_MISSING_FROM_MAP,
+          R.string.feedback_description_road_is_missing_from_map));
+        break;
+      default:
+    }
+    return list;
+  }
+
+  @IntDef({FEEDBACK_MAIN_FLOW, FEEDBACK_DETAIL_FLOW})
+  @Retention(RetentionPolicy.RUNTIME)
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java:469]: Should this be `@Retention(RetentionPolicy.SOURCE)` 🤔?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 5]
```diff
@@ -660,13 +660,14 @@ constructor(
             @FeedbackEvent.Type feedbackType: String,
             description: String,
             @FeedbackEvent.Source feedbackSource: String,
-            screenshot: String?
-        ) {
-            MapboxNavigationTelemetry.postUserFeedback(
+            screenshot: String?,
+            feedbackSubType: Array<String>? = emptyArray()
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:664]: For documentation purposes noting that this breaks SemVer cc @langsmith

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java [hunk 6]
```diff
@@ -287,8 +331,145 @@ public void onFeedbackItemClick(ImageView view, int feedbackPosition) {
...
+          selectedFeedbackItem.getFeedbackSubType().remove(item.getFeedbackDescription());
+          item.setChecked(false);
+          return false;
+        }
+      }
+    };
+
+  private List<FeedbackDescriptionItem> buildFeedbackIssueDetailList(FeedbackItem feedbackItem) {
+    List<FeedbackDescriptionItem> list = new ArrayList<>();
+    switch (feedbackItem.getFeedbackType()) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackBottomSheet.java:383]: NIT I'm not a big fan of `switch`, could we refactor this long one into a map / dictionary?

**Pull request merged**